trieste/acquisition/rule.py

```diff
@@ -93,7 +97,7 @@ class EfficientGlobalOptimization(AcquisitionRule[None, SP_contra]):
 
     def __init__(
         self,
-        builder: AcquisitionFunctionBuilder | None = None,
+        builder: AcquisitionFunctionBuilder | GreedyAcquisitionFunctionBuilder | None = None,
```

Comment:
You don't need this change, since a GreedyAcquisitionFunctionBuilder is a type of AcquisitionFunctionBuilder.

```suggestion
        builder: AcquisitionFunctionBuilder | None = None,
```